### PR TITLE
Support parallel migration task

### PR DESCRIPTION
**Pull request by eccyan**
Hi :)  This PR is supporting migration in parallel.
#75 was using threads, but this PR migrate processes in parallel.
In my environment, the speed got 66% faster  by running the processes in parallel.

__Hardware infomation__

```
$ system_profiler SPHardwareDataType
Hardware:

    Hardware Overview:

      Model Name: MacBook Pro
      Model Identifier: MacBookPro12,1
      Processor Name: Intel Core i5
      Processor Speed: 2.7 GHz
      Number of Processors: 1
      Total Number of Cores: 2
      L2 Cache (per Core): 256 KB
      L3 Cache: 3 MB
      Memory: 16 GB
      Boot ROM Version: MBP121.0167.B17
      SMC Version (system): 2.28f7
      Serial Number (system): C02SC0R2FVH4
      Hardware UUID: B52BB064-8731-5ED4-B78A-48C843E5792A
```

__Speed__

```
bin/rake db:migrate  46.62s user 5.52s system 55% cpu 1:33.62 total
bin/rake db:migrate  43.95s user 5.27s system 54% cpu 1:29.66 total
bin/rake db:migrate  44.61s user 5.37s system 54% cpu 1:30.93 total

bin/rake db:migrate:parallel  66.93s user 7.66s system 131% cpu 56.568 total
bin/rake db:migrate:parallel  67.48s user 7.82s system 132% cpu 57.008 total
bin/rake db:migrate:parallel  66.37s user 7.91s system 131% cpu 56.533 total
```

### CODE REVIEW

**Comment by eccyan** [comment]
@mikecmpbll 
I got up speed in my environment but I am not confident in other environments ...
I don't understand Ruby's threads well.
Does a thread run with smaller overhead in parallel when there are many I/O waits ?

This PR was approved, but if you solve the issue, I will not to merge this PR.

File: spec/tasks/apartment_rake_spec.rb [hunk 1]
```diff
@@ -10,6 +10,7 @@
     load 'tasks/apartment.rake'
     # stub out rails tasks
     Rake::Task.define_task('db:migrate')
+    Rake::Task.define_task('db:migrate:prallel')
```
Comment (mikecmpbll) [inline spec/tasks/apartment_rake_spec.rb]: typo `prallel`

File: spec/tasks/apartment_rake_spec.rb [hunk 2]
```diff
@@ -48,6 +49,17 @@
       end
     end
 
+    describe "apartment:migrate:prallel" do
```
Comment (mikecmpbll) [inline spec/tasks/apartment_rake_spec.rb]: typo `prallel`

**Comment by mikecmpbll** [comment]
@eccyan sorry, it wasn't meant as a criticism, i definitely think we need to support something like this was just trying to get my head around the history of this feature :) 👍 thanks for your work!

**Comment by eccyan** [comment]
@mikecmpbll Oh, thanks 😃 I understood.
We are heavy user of apartment, So I'd like to improve .

**Comment by eccyan** [comment]
@mikecmpbll I fix typo. thanks for review 👍

**Comment by gfx** [comment]
+1

**Comment by michiomochi** [comment]
+1
Please merge this p-r 🙏 (I want to use this feature ☀️  )

**Comment by mikecmpbll** [comment]
i prefer this code to #427 

in order to make it work on public tenant (the problem you explained at https://github.com/influitive/apartment/pull/427#issuecomment-303028690) we could conditionally enhance db:migrate with either apartment:db:migrate or apartment:db:migrate:parallel depending on the configuration of parallel tests.

i will have a stab at this now.

**Comment by mikecmpbll** [comment]
actually, pah, the original shot in #75 does it rather well. i'll get something merged this afternoon.

**Comment by mikecmpbll** [comment]
merged https://github.com/influitive/apartment/pull/493 . thanks for your contribution and patience ❤️

**Comment by gfx** [comment]
@mikecmpbll Can you release a new version? I'd like to use this parallelism.

**Comment by mikecmpbll** [comment]
done! ..

**Pull request closed without merging**